src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ResourceSpecificationKindValueProvider.cs

```diff
@@ -1,47 +1,66 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
 
 using System.Collections.Generic;
+using System.ComponentModel.Composition;
 using System.Threading.Tasks;
 
 namespace Microsoft.VisualStudio.ProjectSystem.Properties
 {
     [ExportInterceptingPropertyValueProvider("ResourceSpecificationKind", ExportInterceptingPropertyValueProviderFile.ProjectFile)]
     internal sealed class ResourceSpecificationKindValueProvider : InterceptingPropertyValueProviderBase
     {
-        // TODO should the rule file generate property and enum value constants that we can use here instead of these string literals?
+        private readonly ITemporaryPropertyStorage _temporaryPropertyStorage;
+
+        private const string Win32ResourceMSBuildProperty = "Win32Resource";
+        private const string ApplicationIconMSBuildProperty = "ApplicationIcon";
+        private const string ApplicationManifestMSBuildProperty = "ApplicationManifest";
+        private const string IconAndManifestValue = "IconAndManifest";
+        private const string ResourceFileValue = "ResourceFile";
+
+        [ImportingConstructor]
+        public ResourceSpecificationKindValueProvider(ITemporaryPropertyStorage temporaryPropertyStorage)
+        {
+            _temporaryPropertyStorage = temporaryPropertyStorage;
+        }
 
         public override async Task<string?> OnSetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties, IReadOnlyDictionary<string, string>? dimensionalConditions = null)
         {
-            if (unevaluatedPropertyValue == "IconAndManifest")
+            if (unevaluatedPropertyValue == IconAndManifestValue)
```

Comment:
>unevaluatedPropertyValue == IconAndManifestValue [](start = 16, length = 48)

Should there be an explicit string comparison specified instead of the == operator?

Reply:
Yes, my bad. Should have been `StringComparers.PropertyLiteralValues` (I believe. See [Teams chat](https://teams.microsoft.com/l/message/19:ba67a1e4184a4f1da5ee42813032ead0@thread.skype/1605754601487?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=4ba7372f-2799-4677-89f0-7a1aaea3706c&parentMessageId=1605754601487&teamName=.NET%20Developer%20Experience&channelName=Project%20System&createdTime=1605754601487) about distinction between `PropertyLiteralValues` and `PropertyValues`).

Reply:
I'll fix that.